core/src/commonMain/kotlin/com/sourcepoint/mobile_core/Coordinator.kt

```diff
@@ -247,16 +255,20 @@ class Coordinator(
         getResponse: ChoiceAllResponse?,
         postResponse: USNatChoiceResponse
     ) {
-        state.usNat?.uuid = postResponse.uuid
-        state.usNat?.applies = state.usNat?.applies?: false
-        state.usNat?.dateCreated = postResponse.dateCreated
-        state.usNat?.expirationDate = postResponse.expirationDate
-        state.usNat?.consentStrings = postResponse.consentStrings
-        state.usNat?.webConsentPayload = postResponse.webConsentPayload?: getResponse?.usnat?.webConsentPayload
-        state.usNat?.userConsents?.categories = postResponse.userConsents.categories
-        state.usNat?.userConsents?.vendors = postResponse.userConsents.vendors
-        state.usNat?.consentStatus = postResponse.consentStatus
-        state.usNat?.gppData = postResponse.gppData
+        state.usNat = state.usNat?.copy(
+            uuid = postResponse.uuid,
+            applies = state.usNat?.applies?: false,
+            dateCreated = postResponse.dateCreated,
+            expirationDate = postResponse.expirationDate,
+            consentStrings = postResponse.consentStrings,
+            webConsentPayload = postResponse.webConsentPayload?: getResponse?.usnat?.webConsentPayload,
+            consentStatus = postResponse.consentStatus,
+            gppData = postResponse.gppData
+        )
+        state.usNat?.userConsents = state.usNat?.userConsents?.copy(
```

Comment:
@Nevazhnovu can `userConsents` be initialized within the `copy` in the line above?